File: ui-android-library/src/main/java/com/xyfindables/ui/views/XYAwesomeButton.java
package com.xyfindables.ui.views;

import android.content.Context;
import android.util.AttributeSet;
import android.view.ContextThemeWrapper;

import com.xyfindables.ui.R;
import com.xyfindables.ui.XYGlobalFonts;

public class XYAwesomeButton extends XYButton {
    public XYAwesomeButton(Context context) {
        this(context, null, 0);
    }

    public XYAwesomeButton(Context context, AttributeSet attrs) {
        this(context, attrs, 0);
    }

    public XYAwesomeButton(Context context, AttributeSet attrs, int defStyle) {
        super(new ContextThemeWrapper(context, R.style.xy_awesome_button), attrs, defStyle);
        XYGlobalFonts.setViewFontAwesome(context, this);
    }
}

File: ui-android-library/src/main/java/com/xyfindables/ui/views/XYAwesomeButton.kt
package com.xyfindables.ui.views

import android.content.Context
import android.util.AttributeSet
import android.view.ContextThemeWrapper

import com.xyfindables.ui.R
import com.xyfindables.ui.XYGlobalFonts

class XYAwesomeButton @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null, defStyle: Int = 0) : XYButton(ContextThemeWrapper(context, R.style.xy_awesome_button), attrs, defStyle) {

    init {
        XYGlobalFonts.setViewFontAwesome(context, this)
    }
}
